platforms/posix/src/lockstep_scheduler/test/src/lockstep_scheduler_test.cpp

```diff
@@ -4,7 +4,43 @@
 #include <atomic>
 #include <random>
 #include <iostream>
+#include <functional>
+#include <chrono>
 
+class TestThread
+{
+public:
+	TestThread(const std::function<void()> &f)
+		: _f(f)
+	{
+		_thread = std::thread(std::bind(&TestThread::execute, this));
+	}
+
+	void join(LockstepScheduler &ls)
+	{
+		// The unit-tests do not reflect the real usage, where
+		// set_absolute_time() is called regularly and can do some
+		// cleanup tasks. We simulate that here by waiting until
+		// the given task returns (which is expected to happen quickly)
+		// and then call set_absolute_time(), which can do the cleanup,
+		// and _thread can then exit as well.
+		while (!_done) {
+			std::this_thread::yield(); // usleep is too slow here
+		}
+
+		ls.set_absolute_time(ls.get_absolute_time());
```

Comment:
Huh, you set the same time again? I don't get it.

Reply:
It's just that the linked list gets cleaned up. Setting the `removed` flag in particular.